app/controllers/sops_controller.rb

```diff
@@ -57,22 +57,35 @@ def update
   end
 
   def dynamic_table_typeahead
-    return if params[:study_id].blank? && params[:assay_id].blank?
+    study_id = params[:study_id] if params[:study_id].present? && !%w[null undefined].include?(params[:study_id].to_s)
+    assay_id = params[:assay_id] if params[:assay_id].present? && !%w[null undefined].include?(params[:assay_id].to_s)
+
+    if study_id.blank? && assay_id.blank?
+      raise "Invalid parameters! Either study id '#{params[:study_id]}' or assay id '#{params[:assay_id]}' must be a valid id."
+    end
 
     query = params[:query] || ''
-    asset = if params[:study_id].present?
-              Study.authorized_for('view').detect { |study| study.id.to_s == params[:study_id] }
+    asset = if study_id.present?
+              study = Study.includes(:sops).find(study_id)
```

Comment:
using `.find` will raise an error if the study_id doesn't exist, whereas `find_by_id` will return nil. Using find_by_id will more closely match the previous behaviour.